src/app/api/users/users.service.ts

```diff
@@ -6,24 +6,23 @@ import { Resource } from '@api/core/types/api'
 import { UserAction, UserRole } from '@api/core/types/user'
 import APIError from '@api/core/exceptions/api'
 import httpStatus from 'http-status'
-import { CopilotListArgs } from '@/types/common'
+import { CopilotListArgs, FilterableUser } from '@/types/common'
+import { filterUsersByKeyword } from '@/utils/users'
 
 class UsersService extends BaseService {
   private copilot: CopilotAPI
+  private DEFAULT_USERS_LIMIT = 10
 
   constructor(user: User) {
     super(user)
     this.copilot = new CopilotAPI(this.user.token)
   }
 
-  async getGroupedUsers() {
+  async getGroupedUsers(limit: number = this.DEFAULT_USERS_LIMIT) {
     const user = this.user
     new PoliciesService(user).authorize(UserAction.Read, Resource.Users)
 
-    const listArgs: CopilotListArgs = {
-      // We want the complete list of users / companies for now
-      limit: 10000,
-    }
+    const listArgs: CopilotListArgs = { limit }
```

Comment:
using the limit to fetch EVERYTHING in one request is still going to cause some problems, the network might take a long time to load and you might have page size limits. What I was suggesting is that we use a large limit 1000 and make multiple calls to the backend to fetch all clients. Its slow, but at least it will always work.

Reply:
Please look into the pagination issue mentioned in Slack